### PR TITLE
Add `ipv6_allowed_for_dual_stack` field for lambda function

### PR DESCRIPTION
**Pull request by DanielRieske**
### Description

This PR adds support for the `ipv6_allowed_for_dual_stack` field on the `aws_lambda_function` resource.

This allows outbound IPv6 traffic on VPC functions that are connected to dual-stack subnets.

Note to the reviewer, I wasn't able to run all tests within the `TestAccLambdaFunction_` suite because of a VPC quota within my project.
 
### Relations

Closes https://github.com/hashicorp/terraform-provider-aws/issues/33922

### Output from Acceptance Testing

```console
terraform-provider-aws git:(f/add-lambda-ipv6-dual-stack-field) ✗ make testacc TESTS=TestAccLambdaFunction_ipv6AllowedForDualStack PKG=lambda
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/lambda/... -v -count 1 -parallel 20 -run='TestAccLambdaFunction_ipv6AllowedForDualStack'  -timeout 360m
=== RUN   TestAccLambdaFunction_ipv6AllowedForDualStack
=== PAUSE TestAccLambdaFunction_ipv6AllowedForDualStack
=== CONT  TestAccLambdaFunction_ipv6AllowedForDualStack
--- PASS: TestAccLambdaFunction_ipv6AllowedForDualStack (784.57s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/lambda     786.716s

...
```

### CODE REVIEW

File: .changelog/34045.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:enhancement
+resource/aws_lambda_function: Added `ipv6_allowed_for_dual_stack` field to support outbound ipv6 traffic on dual-stack subnets
```
Comment (bschaatsbergen) [inline .changelog/34045.txt:2]:
```suggestion
resource/aws_lambda_function: Added `ipv6_allowed_for_dual_stack` argument
```

**Review by bschaatsbergen — APPROVED** [review]
PR overall looks good, thanks for working on this @DanielRieske - I've only placed a comment in regards to the changelog (to keep it as minimal as possible).

Please be sure to lint your acceptance tests 👍🏼. I'll ask @ewbankkit to take a look at this 👀

File: .changelog/34045.txt (hunk 1 above)
Comment (DanielRieske) [reply .changelog/34045.txt]: Absolutely, changed.

**Review by bschaatsbergen — APPROVED** [review]
LGTM 🚀 

`make testacc TESTS=TestAccLambdaFunction_ipv6AllowedForDualStack PKG=lambda                                                                                                                                                                        
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/lambda/... -v -count 1 -parallel 20 -run='TestAccLambdaFunction_ipv6AllowedForDualStack'  -timeout 360m
go: downloading github.com/aws/aws-sdk-go v1.46.0
go: downloading github.com/aws/aws-sdk-go-v2/service/ec2 v1.126.0
go: downloading github.com/aws/aws-sdk-go-v2/service/verifiedpermissions v1.3.0
go: downloading github.com/aws/aws-sdk-go-v2/service/workspaces v1.31.3
=== RUN   TestAccLambdaFunction_ipv6AllowedForDualStack
=== PAUSE TestAccLambdaFunction_ipv6AllowedForDualStack
=== CONT  TestAccLambdaFunction_ipv6AllowedForDualStack
--- PASS: TestAccLambdaFunction_ipv6AllowedForDualStack (803.47s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/lambda	805.679s
`

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccLambdaFunction_' PKG=lambda ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/lambda/... -v -count 1 -parallel 2  -run=TestAccLambdaFunction_ -timeout 360m
=== RUN   TestAccLambdaFunction_basic
=== PAUSE TestAccLambdaFunction_basic
=== RUN   TestAccLambdaFunction_disappears
=== PAUSE TestAccLambdaFunction_disappears
=== RUN   TestAccLambdaFunction_tags
=== PAUSE TestAccLambdaFunction_tags
=== RUN   TestAccLambdaFunction_unpublishedCodeUpdate
=== PAUSE TestAccLambdaFunction_unpublishedCodeUpdate
=== RUN   TestAccLambdaFunction_codeSigning
=== PAUSE TestAccLambdaFunction_codeSigning
=== RUN   TestAccLambdaFunction_concurrency
=== PAUSE TestAccLambdaFunction_concurrency
=== RUN   TestAccLambdaFunction_concurrencyCycle
=== PAUSE TestAccLambdaFunction_concurrencyCycle
=== RUN   TestAccLambdaFunction_expectFilenameAndS3Attributes
=== PAUSE TestAccLambdaFunction_expectFilenameAndS3Attributes
=== RUN   TestAccLambdaFunction_envVariables
=== PAUSE TestAccLambdaFunction_envVariables
=== RUN   TestAccLambdaFunction_EnvironmentVariables_noValue
=== PAUSE TestAccLambdaFunction_EnvironmentVariables_noValue
=== RUN   TestAccLambdaFunction_EnvironmentVariables_emptyUpgrade
=== PAUSE TestAccLambdaFunction_EnvironmentVariables_emptyUpgrade
=== RUN   TestAccLambdaFunction_encryptedEnvVariables
=== PAUSE TestAccLambdaFunction_encryptedEnvVariables
=== RUN   TestAccLambdaFunction_nameValidation
=== PAUSE TestAccLambdaFunction_nameValidation
=== RUN   TestAccLambdaFunction_versioned
=== PAUSE TestAccLambdaFunction_versioned
=== RUN   TestAccLambdaFunction_versionedUpdate
=== PAUSE TestAccLambdaFunction_versionedUpdate
=== RUN   TestAccLambdaFunction_enablePublish
=== PAUSE TestAccLambdaFunction_enablePublish
=== RUN   TestAccLambdaFunction_disablePublish
=== PAUSE TestAccLambdaFunction_disablePublish
=== RUN   TestAccLambdaFunction_deadLetter
=== PAUSE TestAccLambdaFunction_deadLetter
=== RUN   TestAccLambdaFunction_deadLetterUpdated
=== PAUSE TestAccLambdaFunction_deadLetterUpdated
=== RUN   TestAccLambdaFunction_nilDeadLetter
=== PAUSE TestAccLambdaFunction_nilDeadLetter
=== RUN   TestAccLambdaFunction_fileSystem
=== PAUSE TestAccLambdaFunction_fileSystem
=== RUN   TestAccLambdaFunction_image
    function_test.go:979: Environment variable AWS_LAMBDA_IMAGE_LATEST_ID is not set
--- SKIP: TestAccLambdaFunction_image (0.00s)
=== RUN   TestAccLambdaFunction_architectures
=== PAUSE TestAccLambdaFunction_architectures
=== RUN   TestAccLambdaFunction_architecturesUpdate
=== PAUSE TestAccLambdaFunction_architecturesUpdate
=== RUN   TestAccLambdaFunction_architecturesWithLayer
=== PAUSE TestAccLambdaFunction_architecturesWithLayer
=== RUN   TestAccLambdaFunction_ephemeralStorage
=== PAUSE TestAccLambdaFunction_ephemeralStorage
=== RUN   TestAccLambdaFunction_tracing
=== PAUSE TestAccLambdaFunction_tracing
=== RUN   TestAccLambdaFunction_KMSKeyARN_noEnvironmentVariables
=== PAUSE TestAccLambdaFunction_KMSKeyARN_noEnvironmentVariables
=== RUN   TestAccLambdaFunction_layers
=== PAUSE TestAccLambdaFunction_layers
=== RUN   TestAccLambdaFunction_layersUpdate
=== PAUSE TestAccLambdaFunction_layersUpdate
=== RUN   TestAccLambdaFunction_vpc
=== PAUSE TestAccLambdaFunction_vpc
=== RUN   TestAccLambdaFunction_vpcRemoval
=== PAUSE TestAccLambdaFunction_vpcRemoval
=== RUN   TestAccLambdaFunction_vpcUpdate
=== PAUSE TestAccLambdaFunction_vpcUpdate
=== RUN   TestAccLambdaFunction_VPC_withInvocation
=== PAUSE TestAccLambdaFunction_VPC_withInvocation
=== RUN   TestAccLambdaFunction_VPCPublishNo_changes
=== PAUSE TestAccLambdaFunction_VPCPublishNo_changes
=== RUN   TestAccLambdaFunction_VPCPublishHas_changes
=== PAUSE TestAccLambdaFunction_VPCPublishHas_changes
=== RUN   TestAccLambdaFunction_VPC_properIAMDependencies
=== PAUSE TestAccLambdaFunction_VPC_properIAMDependencies
=== RUN   TestAccLambdaFunction_VPC_replaceSGWithDefault
=== PAUSE TestAccLambdaFunction_VPC_replaceSGWithDefault
=== RUN   TestAccLambdaFunction_VPC_replaceSGWithCustom
=== PAUSE TestAccLambdaFunction_VPC_replaceSGWithCustom
=== RUN   TestAccLambdaFunction_emptyVPC
=== PAUSE TestAccLambdaFunction_emptyVPC
=== RUN   TestAccLambdaFunction_s3
=== PAUSE TestAccLambdaFunction_s3
=== RUN   TestAccLambdaFunction_localUpdate
=== PAUSE TestAccLambdaFunction_localUpdate
=== RUN   TestAccLambdaFunction_LocalUpdate_nameOnly
=== PAUSE TestAccLambdaFunction_LocalUpdate_nameOnly
=== RUN   TestAccLambdaFunction_S3Update_basic
=== PAUSE TestAccLambdaFunction_S3Update_basic
=== RUN   TestAccLambdaFunction_S3Update_unversioned
=== PAUSE TestAccLambdaFunction_S3Update_unversioned
=== RUN   TestAccLambdaFunction_snapStart
=== PAUSE TestAccLambdaFunction_snapStart
=== RUN   TestAccLambdaFunction_runtimes
=== PAUSE TestAccLambdaFunction_runtimes
=== RUN   TestAccLambdaFunction_Zip_validation
=== PAUSE TestAccLambdaFunction_Zip_validation
=== RUN   TestAccLambdaFunction_ipv6AllowedForDualStack
=== PAUSE TestAccLambdaFunction_ipv6AllowedForDualStack
=== RUN   TestAccLambdaFunction_skipDestroy
=== PAUSE TestAccLambdaFunction_skipDestroy
=== RUN   TestAccLambdaFunction_skipDestroyInconsistentPlan
=== PAUSE TestAccLambdaFunction_skipDestroyInconsistentPlan
=== CONT  TestAccLambdaFunction_basic
=== CONT  TestAccLambdaFunction_tracing
--- PASS: TestAccLambdaFunction_basic (43.29s)
=== CONT  TestAccLambdaFunction_emptyVPC
--- PASS: TestAccLambdaFunction_tracing (74.61s)
=== CONT  TestAccLambdaFunction_skipDestroyInconsistentPlan
--- PASS: TestAccLambdaFunction_skipDestroyInconsistentPlan (107.48s)
=== CONT  TestAccLambdaFunction_skipDestroy
--- PASS: TestAccLambdaFunction_emptyVPC (44.22s)
=== CONT  TestAccLambdaFunction_ipv6AllowedForDualStack
--- PASS: TestAccLambdaFunction_skipDestroy (37.51s)
=== CONT  TestAccLambdaFunction_Zip_validation
--- PASS: TestAccLambdaFunction_Zip_validation (5.33s)
=== CONT  TestAccLambdaFunction_runtimes
--- PASS: TestAccLambdaFunction_runtimes (742.41s)
=== CONT  TestAccLambdaFunction_snapStart
--- PASS: TestAccLambdaFunction_ipv6AllowedForDualStack (816.05s)
=== CONT  TestAccLambdaFunction_VPC_withInvocation
--- PASS: TestAccLambdaFunction_snapStart (67.40s)
=== CONT  TestAccLambdaFunction_S3Update_unversioned
--- PASS: TestAccLambdaFunction_S3Update_unversioned (90.48s)
=== CONT  TestAccLambdaFunction_VPC_replaceSGWithCustom
--- PASS: TestAccLambdaFunction_VPC_replaceSGWithCustom (358.22s)
=== CONT  TestAccLambdaFunction_VPC_replaceSGWithDefault
--- PASS: TestAccLambdaFunction_VPC_replaceSGWithDefault (254.63s)
=== CONT  TestAccLambdaFunction_S3Update_basic
--- PASS: TestAccLambdaFunction_S3Update_basic (82.66s)
=== CONT  TestAccLambdaFunction_LocalUpdate_nameOnly
--- PASS: TestAccLambdaFunction_LocalUpdate_nameOnly (61.83s)
=== CONT  TestAccLambdaFunction_localUpdate
--- PASS: TestAccLambdaFunction_localUpdate (78.42s)
=== CONT  TestAccLambdaFunction_s3
--- PASS: TestAccLambdaFunction_s3 (59.32s)
=== CONT  TestAccLambdaFunction_versioned
--- PASS: TestAccLambdaFunction_versioned (41.45s)
=== CONT  TestAccLambdaFunction_VPC_properIAMDependencies
--- PASS: TestAccLambdaFunction_VPC_properIAMDependencies (263.65s)
=== CONT  TestAccLambdaFunction_ephemeralStorage
--- PASS: TestAccLambdaFunction_ephemeralStorage (81.25s)
=== CONT  TestAccLambdaFunction_VPCPublishHas_changes
--- PASS: TestAccLambdaFunction_VPC_withInvocation (1582.76s)
=== CONT  TestAccLambdaFunction_architecturesWithLayer
--- PASS: TestAccLambdaFunction_architecturesWithLayer (74.15s)
=== CONT  TestAccLambdaFunction_architecturesUpdate
--- PASS: TestAccLambdaFunction_architecturesUpdate (67.42s)
=== CONT  TestAccLambdaFunction_architectures
--- PASS: TestAccLambdaFunction_architectures (59.09s)
=== CONT  TestAccLambdaFunction_fileSystem
--- PASS: TestAccLambdaFunction_fileSystem (998.65s)
=== CONT  TestAccLambdaFunction_VPCPublishNo_changes
--- PASS: TestAccLambdaFunction_VPCPublishHas_changes (1717.57s)
=== CONT  TestAccLambdaFunction_nilDeadLetter
--- PASS: TestAccLambdaFunction_nilDeadLetter (61.05s)
=== CONT  TestAccLambdaFunction_deadLetterUpdated
--- PASS: TestAccLambdaFunction_deadLetterUpdated (65.38s)
=== CONT  TestAccLambdaFunction_expectFilenameAndS3Attributes
--- PASS: TestAccLambdaFunction_expectFilenameAndS3Attributes (1.82s)
=== CONT  TestAccLambdaFunction_deadLetter
--- PASS: TestAccLambdaFunction_deadLetter (64.97s)
=== CONT  TestAccLambdaFunction_disablePublish
--- PASS: TestAccLambdaFunction_disablePublish (66.90s)
=== CONT  TestAccLambdaFunction_enablePublish
--- PASS: TestAccLambdaFunction_VPCPublishNo_changes (470.65s)
=== CONT  TestAccLambdaFunction_versionedUpdate
--- PASS: TestAccLambdaFunction_enablePublish (83.88s)
=== CONT  TestAccLambdaFunction_nameValidation
--- PASS: TestAccLambdaFunction_nameValidation (1.84s)
=== CONT  TestAccLambdaFunction_codeSigning
--- PASS: TestAccLambdaFunction_versionedUpdate (102.23s)
=== CONT  TestAccLambdaFunction_encryptedEnvVariables
--- PASS: TestAccLambdaFunction_codeSigning (78.99s)
=== CONT  TestAccLambdaFunction_concurrencyCycle
--- PASS: TestAccLambdaFunction_concurrencyCycle (80.84s)
=== CONT  TestAccLambdaFunction_EnvironmentVariables_emptyUpgrade
--- PASS: TestAccLambdaFunction_EnvironmentVariables_emptyUpgrade (141.97s)
=== CONT  TestAccLambdaFunction_concurrency
--- PASS: TestAccLambdaFunction_encryptedEnvVariables (103.60s)
=== CONT  TestAccLambdaFunction_EnvironmentVariables_noValue
--- PASS: TestAccLambdaFunction_concurrency (60.10s)
=== CONT  TestAccLambdaFunction_vpc
--- PASS: TestAccLambdaFunction_EnvironmentVariables_noValue (49.66s)
=== CONT  TestAccLambdaFunction_envVariables
--- PASS: TestAccLambdaFunction_vpc (1769.04s)
=== CONT  TestAccLambdaFunction_vpcUpdate
--- PASS: TestAccLambdaFunction_vpcUpdate (653.77s)
=== CONT  TestAccLambdaFunction_vpcRemoval
--- PASS: TestAccLambdaFunction_envVariables (112.32s)
=== CONT  TestAccLambdaFunction_layers
--- PASS: TestAccLambdaFunction_layers (174.80s)
=== CONT  TestAccLambdaFunction_layersUpdate
--- PASS: TestAccLambdaFunction_layersUpdate (71.04s)
=== CONT  TestAccLambdaFunction_tags
--- PASS: TestAccLambdaFunction_tags (76.59s)
=== CONT  TestAccLambdaFunction_unpublishedCodeUpdate
--- PASS: TestAccLambdaFunction_unpublishedCodeUpdate (63.91s)
=== CONT  TestAccLambdaFunction_disappears
--- PASS: TestAccLambdaFunction_disappears (37.35s)
=== CONT  TestAccLambdaFunction_KMSKeyARN_noEnvironmentVariables
--- PASS: TestAccLambdaFunction_vpcRemoval (486.60s)
--- PASS: TestAccLambdaFunction_KMSKeyARN_noEnvironmentVariables (41.50s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/lambda	4994.889s
```

**Pull request merged**